integration-tests/webhook/webhook_test.go

```diff
@@ -0,0 +1,163 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: demo-snap-vol-claim
+  annotations:
+    snapshot.alpha.kubernetes.io/snapshot: fastfurious
+spec:
+  storageClassName: openebs-snapshot-promoter
+  accessModes: [ "ReadWriteOnce" ]
+  resources:
+    requests:
+      storage: 2G`
+)
+
+var _ = Describe("AdmissionWebhook", func() {
+	BeforeEach(func() {
+		// Extracting storageclass artifacts unstructured
+		_, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(artifacts.SingleReplicaSC),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting PVC artifacts unstructured
+		PVCUnst, err := artifacts.GetArtifactUnstructured(artifacts.Artifact(artifacts.CStorPVCArtifacts))
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting statefulset application namespace
+		PVCNamespace := PVCUnst.GetNamespace()
+
+		// Webhook stuffs
+		client, err := k8s.Clientset().Get()
+		client.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().List(metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			ginkgo.Skip("dynamic configuration of webhooks requires the admissionregistration.k8s.io group to be enabled")
+		}
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Create pvc using storageclass 'cstor-sparse-pool'
+		cu := k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(PVCUnst),
+			PVCUnst.GetNamespace(),
+		)
+		_, err = cu.Apply(PVCUnst)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Check for pvc to get created and bound
+		Eventually(func() string {
+			pvclaim, err := pvc.
+				KubeClient(pvc.WithNamespace(PVCNamespace)).Get(PVCNamespace, PVCUnst.GetName())
+			Expect(err).ShouldNot(HaveOccurred())
+			return string(pvclaim.Status.Phase)
+		},
+			defaultTimeOut, defaultPollingInterval).
+			Should(Receive((ContainSubstring("Bound")), "Pvc phase should bound"))
+
+		PVLabel := "openebs.io/casType=" + "cstor"
+		// Check for CVR to get healthy
+		Eventually(func() int {
+			cvrs, err := cvr.
+				KubeClient(cvr.WithNamespace("")).
+				List(metav1.ListOptions{LabelSelector: PVLabel})
+			Expect(err).ShouldNot(HaveOccurred())
+			return cvr.
+				ListBuilder().
+				WithAPIList(cvrs).
+				WithFilter(cvr.IsHealthy()).
+				List().
+				Len()
+		},
+			defaultTimeOut, defaultPollingInterval).
+			Should(Equal(1), "CVR count should be "+string(1))
+
+	})
+
+	AfterEach(func() {
+
+		// Extracting statefulset artifacts unstructured
+		PVCUnst, err := artifacts.GetArtifactUnstructured(artifacts.Artifact(artifacts.CStorPVCArtifacts))
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting pvc namespace
+		PVCNamespace := PVCUnst.GetNamespace()
+
+		// Generating label selector for stsResources
+		PVCLabel := "openebs.io/casType=" + "cstor"
+
+		// Fetch PVCs to be deleted
+		pvcs, err := pvc.KubeClient(pvc.WithNamespace(PVCNamespace)).
+			List(metav1.ListOptions{LabelSelector: PVCLabel})
+		Expect(err).ShouldNot(HaveOccurred())
+		// Delete PVCs
+		for _, p := range pvcs.Items {
```

Comment:
Try to avoid for loop & make use of `DeleteCollection`

Reply:
Not required this code.. but its good to have `DeleteCollection` to delete all in  BDD.
Will add them in upcoming PR's